### PR TITLE
[FLOSS H3] [FLOSS T2] Make Elektra reuse compliant, add check in Jenkins pipeline

### LINKED ISSUE #4094: reuse lint
[`reuse lint`](https://reuse.software/) complains:

* Bad licenses: BSD-0-clause, BSD-2-clause-discount, BSD-3-clause, BSD-3-clause-cmake, BSD-3-clause-cmake-new, BSD-3-clause-markdownconverter, public-domain
* Missing licenses: BSD-0-clause, BSD-2-Clause-FreeBSD, BSD-2-clause-discount, BSD-3-clause, BSD-3-clause-cmake, BSD-3-clause-cmake-new, BSD-3-clause-markdownconverter, public-domain

some licenses already in .reuse/dep5 are missing in LICENSES

### PR DESCRIPTION
**Pull request by sedadas**
Closes #4094

## Basics

These points need to be fulfilled for every PR:

- [x] Short descriptions of your changes are in the release notes
      (added as entry in `doc/news/_preparation_next_release.md` which
      contains `_(my name)_`)
      **Please always add something to the release notes.**
- [x] Details of what you changed are in commit messages
      (first line should have `module: short statement` syntax)
- [x] References to issues, e.g. `close #X`, are in the commit messages.
- [ ] The buildservers are happy. If not, fix **in this order**:
  - [ ] add a line in `doc/news/_preparation_next_release.md`
  - [ ] reformat the code with `scripts/dev/reformat-all`
  - [ ] make all unit tests pass
  - [ ] fix all memleaks
- [x] The PR is rebased with current master.

If you have any troubles fulfilling these criteria, please write
about the trouble as comment in the PR. We will help you.
But we cannot accept PRs that do not fulfill the basics.

## Checklist

Check relevant points but **please do not remove entries**.
For docu fixes, spell checking, and similar none of these points below
need to be checked.

- [ ] I added unit tests for my code
- [ ] I fully described what my PR does in the documentation
      (not in the PR description)
- [ ] I fixed all affected documentation
- [ ] I added code comments, logging, and assertions as appropriate (see [Coding Guidelines](https://master.libelektra.org/doc/CODING.md))
- [ ] I updated all meta data (e.g. README.md of plugins and [METADATA.ini](https://master.libelektra.org/doc/METADATA.ini))
- [ ] I mentioned every code not directly written by me in [THIRD-PARTY-LICENSES](doc/THIRD-PARTY-LICENSES)

## Review

Reviewers will usually check the following:

- [ ] Documentation is introductory, concise, good to read and describes everything what the PR does
- [ ] Examples are well chosen and understandable
- [ ] Code is conforming to [our Coding Guidelines](https://master.libelektra.org/doc/CODING.md)
- [ ] APIs are conforming to [our Design Guidelines](https://master.libelektra.org/doc/DESIGN.md)
- [ ] Code is consistent to [our Design Decisions](https://master.libelektra.org/doc/decisions)

## Labels

If you are already Elektra developer:

- Add the "work in progress" label if you do not want the PR to be reviewed yet.
- Add the "ready to merge" label **if the basics are fulfilled** and you also
  say that everything is ready to be merged.

### CODE REVIEW

**Comment by sedadas** [comment]
@markus2330 currently, I get the following output:

```
$ reuse lint
# DEPRECATED LICENSES

The following licenses are deprecated by SPDX:
* BSD-2-Clause-FreeBSD


# SUMMARY

* Bad licenses:
* Deprecated licenses: BSD-2-Clause-FreeBSD
* Licenses without file extension:
* Missing licenses:
* Unused licenses:
* Used licenses: Apache-2.0, BSD-2-Clause-FreeBSD, BSD-3-Clause, BSL-1.0, LicenseRef-BSD-0-clause, LicenseRef-BSD-2-clause-discount, LicenseRef-BSD-3-clause, LicenseRef-BSD-3-clause-cmake, LicenseRef-BSD-3-clause-cmake-new, LicenseRef-BSD-3-clause-markdownconverter, LicenseRef-public-domain, MIT, OFL-1.1, Zlib
* Read errors: 0
* Files with copyright information: 3066 / 3066
* Files with license information: 3066 / 3066

Unfortunately, your project is not compliant with version 3.0 of the REUSE Specification :-(
```

I suspect that even though the references have been added, the tool complains because `BSD-2-Clause-FreeBSD` has been deprecated and superseded by the `BSD-2-Clause-Views` license. See:

https://spdx.org/licenses/BSD-2-Clause-FreeBSD.html
https://spdx.org/licenses/BSD-2-Clause-Views.html

The license is used under `src/libs/version_gen.awk`

**Review by flo91 — COMMENTED** [review]
**[FLOSS H3]**

Thanks for the useful changes and additions in this PR!

I ran the `reuse lint` command on the current master branch and on the branch of your PR and can confirm the results that you got and described in the issue #4094 and this PR.

I suggest to wait for input from @markus2330 to decide if the deprecated `BSD-2-Clause-FreeBSD` should be changed.

Regarding the commits, it may is a good idea to combine ("squash") your two commits into one and add a module-prefix to fulfil the format as specified by the PR-description.

>  Details of what you changed are in commit messages
> (first line should have `module: short statement` syntax)

Currently, one buildserver-check is failing because of a changed manpage, but that should probably get fixed by rebasing you PR with the current master branch.

Good work! 👍🏻

**Comment by markus2330** [comment]
> I suspect that even though the references have been added, the tool complains because BSD-2-Clause-FreeBSD has been deprecated and superseded by the BSD-2-Clause-Views license.

I suspect the same. It looks to me as BSD-2-Clause-FreeBSD is actually the wrong identifier (the text never used "FreeBSD" in the text as it would have been required for BSD-2-Clause-FreeBSD) and https://spdx.org/licenses/BSD-2-Clause-Views.html refers to the license we actually use.

File: LICENSES/BSD-2-Clause-FreeBSD.txt [hunk 1]
```diff
@@ -0,0 +1,20 @@
+Redistribution and use in source and binary forms, with or without
```
Comment (markus2330) [inline LICENSES/BSD-2-Clause-FreeBSD.txt:1]: not needed

File: doc/news/_preparation_next_release.md [hunk 2]
```diff
@@ -223,6 +223,10 @@ plugins, bindings and tools are always up to date. Furthermore, we changed:
 - Remove links to Travis CI and replace them with Github Actions (with badge). _(Mihael Pranjić)_
 - <<TODO>>
 
+## Other
+
+- Added missing licenses and license references _(Ivaylo Ivanov)_
```
Comment (markus2330) [inline doc/news/_preparation_next_release.md:228]: You made it now finally reuse compliant!

**Review by markus2330 — CHANGES_REQUESTED** [review]
Excellent job! :fireworks:  Can you also add a build server check? (T2) :rocket:

File: scripts/jenkins/Jenkinsfile [hunk 3]
```diff
@@ -838,6 +841,19 @@ def buildCheckReleaseNotes() {
   }]
 }
 
+/* Stage checking if for reuse compliancy */
+def buildCheckReuseCompliant() {
+  def stageName = "check-reuse-compliant"
+  return [(stageName): {
+    maybeStage(stageName, !isMaster()) {
+      withDockerEnv(DOCKER_IMAGES.bullseye, [DockerOpts.MOUNT_MIRROR]) {
```
Comment (markus2330) [inline scripts/jenkins/Jenkinsfile:849]: make sure that you also install reuse in the bullseye container: scripts/docker/debian/bullseye/Dockerfile

Comment (sedadas) [reply scripts/jenkins/Jenkinsfile:849]: nice catch. Installation has been added.

**Comment by markus2330** [comment]
scripts/cmake/Modules/COPYING-CMAKE-SCRIPTS and ./src/plugins/ni/nickel-1.1.0/COPYING seems to be a duplicate licenses which actually should be in LICENSES. Can you please remove that files in this PR?

**Comment by sedadas** [comment]
@markus2330 the files have been removed

**Comment by markus2330** [comment]
Amazing job! :1st_place_medal:

**Pull request merged**